pkg/client/cli/cmd/quit.go

```diff
@@ -1,52 +1,32 @@
 package cmd
 
 import (
-	"os"
-
 	"github.com/spf13/cobra"
 
 	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/ann"
 	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/connect"
-	"github.com/telepresenceio/telepresence/v2/pkg/ioutil"
 )
 
 func quit() *cobra.Command {
 	quitDaemons := false
-	quitRootDaemon := false
-	quitUserDaemon := false
 	cmd := &cobra.Command{
-		Use:  "quit",
-		Args: cobra.NoArgs,
-
-		Short:       "Tell telepresence daemon to quit",
-		Annotations: map[string]string{ann.UserDaemon: ann.Optional},
+		Use:   "quit",
+		Args:  cobra.NoArgs,
+		Short: "Tell telepresence daemons to quit",
 		RunE: func(cmd *cobra.Command, _ []string) error {
-			if err := connect.InitCommand(cmd); err != nil {
-				return err
-			}
-			if quitUserDaemon {
-				ioutil.Println(os.Stderr, "--user-daemon (-u) is deprecated, please use --stop-daemons (-s)")
-				quitDaemons = true
-			}
-			if quitRootDaemon {
-				ioutil.Println(os.Stderr, "--root-daemon (-r) is deprecated, please use --stop-daemons (-s)")
-				quitDaemons = true
-			}
 			if quitDaemons {
 				connect.Quit(cmd.Context())
 			} else {
+				cmd.Annotations = map[string]string{ann.UserDaemon: ann.Optional}
+				if err := connect.InitCommand(cmd); err != nil {
+					return err
+				}
 				connect.Disconnect(cmd.Context())
 			}
 			return nil
 		},
 	}
 	flags := cmd.Flags()
 	flags.BoolVarP(&quitDaemons, "stop-daemons", "s", false, "stop all local telepresence daemons")
-	flags.BoolVarP(&quitRootDaemon, "root-daemon", "r", false, "stop daemons")
```

Comment:
Whenever we want to deprecate a flag, we ensure that a deprecation warning is printed. We then wait at least two minor releases before removing it. So a flag deprecated in 2.17.x will not be removed until 2.20.x.

These two flags have been deprecated since version 2.8.0 (released October last year) .